### PR TITLE
integrated the use of the leaderboard API

### PR DESCRIPTION
**Pull request by CobbyElsonfx**
### In this PR I implemented the following🎄:

- [ ] Integrated the Leaderboard API
- [ ] Made use of Async await 
- [ ] Categorized my code into simple blocks of modules
- [ ] Added a function to make a post request
- [ ] Added a function to retrieve all data using a get request method

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -3,4 +3,20 @@ import './assets/github.svg';
...
+
+  const form = document.getElementById('form');
+  const user = form.user.value;
+  const score = form.score.value;
+  postScores(user, score);
+  form.reset();
+});
+
+const refreshBtn = document.querySelector('#refreshBtn');
+refreshBtn.onclick = displayData;
```
Comment (Strangeal) [inline src/index.js]: - [ ] Excellent work on achieving the API integration milestone! However, during my review, I observed that the scores are not being fetched on page load; they are only retrieved when the user clicks the `refresh button.` This behavior is not intended, and we need the scores to be fetched automatically without requiring the user to click anything.

   - To resolve this issue, I suggest adding a line of code to make use of the `window.onload` event to trigger the `displayData` function when the page finishes loading. By doing so, the scores will be fetched and displayed as soon as the page is ready, providing the desired behavior.

File: src/modules/postScores.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      method: 'post',
+      body: JSON.stringify(userData),
+      headers: {
+        'Content-type': 'application/json; charset=UTF-8',
+      },
+    });
+    if (data.status === 201) {
+      const successMess = document.querySelector('.successMess');
+      successMess.textContent = 'Score added successfully';
+    }
```
Comment (Strangeal) [inline src/modules/postScores.js]: - [ ] Excellent work on displaying the success message when the score is created. Your implementation shows promise! However, I believe there's room for improvement by incorporating a `setTimeout()` function.

   - Using the `setTimeout()` function will enhance the user experience by controlling how long the message remains visible on the UI. You can specify a desired time interval for the message to appear and then automatically disappear after the set duration.
   - By incorporating `setTimeout()`, you can tailor the message display time as needed and ensure that it does not clutter the UI.

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @CobbyElsonfx, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 
- Professional `README` template ✅ 
- Correct `GitHub` workflow ✅ 
- Meaningful **Github** commits ✅  
- Correct `.github/workflows` for linters ✅ 
- Followed *best-practices* ✅ 
- Correct **Semantics** used ✅ 
- Well structured project ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by abdulhamiid — APPROVED** [review]
# :green_circle:  Approved! :partying_face:
Hi @CobbyElsonfx 👋, 
image
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🌟

- ✔️ Descriptions are very descriptive
- ✔️ Send and receive data from an API.
- ✔️ Understand and use JSON.

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**